File: j_orca/백준/Bronze/11050. 이항 계수 1/이항 계수 1.js
const fs = require('fs');

const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';

let input = fs.readFileSync(filePath).toString().trim().split('\n');

solution(input);

function solution(input) {
  const [N, K] = input[0].split(' ').map(Number);
  let acc = 1;
  const cache = Array.from({length: N + 1}, (_, i) => i === 0 ? acc : acc *= (i));
  const answer = cache[N] / (cache[N-K] * cache[K]);
  console.log(answer);
} 

File: hahahaeun/백준/Bronze/11050. 이항 계수 1/이항 계수 1.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())

def binomial(x):
    if x == 0: 
        return 1
    return x * binomial(x-1)
    

print(binomial(n)//(binomial(k)*binomial(n-k)))
